src/server/twemcache-rs/src/protocol/data.rs

```diff
@@ -0,0 +1,1023 @@
+// Copyright 2021 Twitter, Inc.
```

Comment:
I assume there's also a plan for modularizing protocol? If so I'll skip for now :P

Reply:
Yes. This requires some thought to modularize. I still want you to look over parsing in the data protocol.

Reply:
The abstraction doesn't look quite right. This isn't a pure protocol module as-is, it's mixing parsing with processing. I think it's harder to deal with for memcached than RESP/redis, but it should be achievable.

The bridge between the two is something that looks like a request object that has no serialization/protocol syntax mixed it. That was what I tried to create in the C version. So anytime we see a string literal it should probably be in the parser, and otherwise in the server.

Reply:
Somewhat better now =)